src/hotspot/share/opto/callGenerator.cpp

```diff
@@ -648,6 +648,9 @@ void CallGenerator::do_late_inline_helper() {
       map->init_req(i1, call->in(i1));
     }
 
+    // Clear the allocation state. We assume all inputs are materialized.
+    jvms->alloc_state().clear();
```

Comment:
@navyxliu any ideas if there is better place to put this?

Reply:
I think `CallGenerator::do_late_inline_helper` is okay. 
it's common code for all late inliner callsites.

Reply:
Can we clear it right after `JVMState* jvms = old_jvms->clone_shallow(C);`? it should be only a cosmic change.  

now you put this statement in the middle of map initialization.

Reply:
Done